utils/src/field_helpers.rs

```diff
@@ -95,6 +95,14 @@ pub fn i32_to_field<F: From<u64> + Neg<Output = F>>(i: i32) -> F {
     }
 }
 
+pub fn i128_to_field<F: From<u128> + Neg<Output = F>>(i: i128) -> F {
+    if i >= 0 {
+        F::from(i as u128)
+    } else {
+        -F::from(-i as u128)
```

Comment:
this still looks dangerous to me :o casting `i128` to `u128`

Reply:
The signed version should be about half the size of the unsigned version, but to be safe maybe use the `From` or `TryFrom` trait?

Example `-F::from(u128::from(i))`

Or guard with `std::u128::MAX`

Reply:
there are no From implementations usually when the conversion can be dangerous

Reply:
you could use TryFrom and return an error (or panic if you know that this will never be given as a malicious input)

Reply:
Oh this chat is very enlightening... Nonetheless, this edge-case you're talking about is not the one we have here right? There you're saying:
`    let a = -i128::MAX;
    let b = a as u128;`
And then `a = -170141183460469231731687303715884105727`
and `b=170141183460469231731687303715884105729` 

But the conditional in the helper is instead doing this:
`    let a = -i128::MAX;
    let b = -a as u128;`
And then `a = -170141183460469231731687303715884105727`
and `b=170141183460469231731687303715884105727` 

Let's just use some smaller data so it is easier to imagine what's going on:
`let a = -i8::MAX;`
`let b = a as u8;`

In this case, `a = 0b1000_0001 as i8 => -127` and `b = 0b1000_0001 as u8 =>129`

But if we perform `b = -a as u8` , first, it negates obtaining +127 and then 127 is expressed as u8 with `0b0111_1111`.

Nonetheless, I think what you meant instead is the edge-case where `a=-i8::MIN`, meaning -128. In that case, performing `b=-a as u8` effectively tries to perform -(-128) which does not fit inside a `u8`, even if later on +128 fits inside `u8` just fine. 

So you're right! There's that issue when you try to convert the minimum value of the integer range. I am using this helper only to convert a bytecode program to field elements in the tests, without having the user write directly field elements, and I was not using `i128::MIN` for that. But if a helper is meant to work in any scenario, then I agree that this must be fixed or avoided somehow. Nonetheless, this should be upgraded so that indeed one could write field elements directly, as in the playground. I will fix this in the next commit of the pr.

Reply:
oh you're right, I actually didn't think of the edge case properly. So indeed this causes an integer overflow:

```rust
let a = i128::MIN;
let b = -a as u128;
```

 https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5770463ba987cd28eb7b0647fa397931

Reply:
makes me think that the function above (`i32_to_field`) could also be simplified as a one-liner `F::from(i.abs() as u64)`